addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1963,15 +1966,19 @@ export class OdooEditor extends EventTarget {
             // <font> node. If it remains empty, it will be cleaned up later by
             // the sanitizer.
             const zws = document.createTextNode('\u200B');
-            range.startContainer.before(zws);
+            start.before(zws);
             insertedZws = zws;
         }
-        let { startContainer: start, startOffset, endContainer: end, endOffset } = range;
-        const [startBlock, endBlock] = [closestBlock(start), closestBlock(end)];
+        // Do not join blocks in the following cases:
```

Comment:
In the original commit you only defined `startBlock` and `endBlock` here. Can you do that still?

Reply:
No I can't as i want the `firstLeaf` of `startBlock` and `lastLeaf` of `endBlock`. When the zws will be inserted the firstLeaf would change. Maybe I could define endBlock here, but will that be a good idea to define startBlock above and endBlock below?

Reply:
No you can leave it the way you did it then, thanks.